src/Rules/Register.hs

```diff
@@ -1,44 +1,101 @@
-module Rules.Register (registerPackage) where
+{-# LANGUAGE TupleSections #-}
+module Rules.Register (registerPackages) where
 
 import Base
 import Context
 import GHC
 import Target
 import Utilities
+import Hadrian.Expression
+import Settings
 
--- TODO: Simplify.
+import Distribution.ParseUtils
+import qualified Distribution.Compat.ReadP as Parse
+import Distribution.Version (Version)
+
+import Hadrian.Haskell.Cabal.Parse as Cabal
+
+parseCabalName :: String -> Maybe (String, Version)
+parseCabalName = readPToMaybe parse
+  where parse = (,) <$> (parsePackageName <* Parse.char '-') <*> parseOptVersion
+
+-- | This rule provides rules for copying packges into the
+-- boot packages db from the installed compiler.
```

Comment:
You'll need to either merge this with the haddock comment below or drop it.

Reply:
will do.